src/test/regress/citus_tests/run_test.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+import pathlib
+from glob import glob
+import argparse
+import shutil
+import random
+import re
+import common
+
+args = argparse.ArgumentParser()
+args.add_argument("-t", "--test", required=True, help="Relative path for test file (must have a .sql or .spec extension)", type=pathlib.Path)
+args.add_argument("-n", "--ntimes", required=True, help="Number of test to run", type=int)
+args.add_argument("-s", "--schedule", required=False, help="Test schedule to be used as a base (optional)", nargs='?', const='', default='')
+
+args = vars(args.parse_args())
+
+regress_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+test_file_path = os.path.join(os.getcwd(), args['test'])
+
+if not os.path.isfile(test_file_path):
+    print(f"ERROR: test file '{test_file_path}' does not exist")
+    sys.exit(2)
+
+test_file_extension = pathlib.Path(test_file_path).suffix
+test_file_name = pathlib.Path(test_file_path).stem
+
+if not test_file_extension in '.spec.sql':
+    print(
+        "ERROR: Unrecognized test extension. Valid extensions are: .sql and .spec"
+    )
+    sys.exit(1)
+
+test_schedule = ''
+
+# find related schedule
+for schedule_file_path in sorted(glob(os.path.join(regress_dir, "*_schedule"))):
+        for schedule_line in open(schedule_file_path, 'r'):
+            if  re.search(r'\b' + test_file_name + r'\b', schedule_line):
+                test_schedule = pathlib.Path(schedule_file_path).stem
+                test_schedule_line = schedule_line
+                break
+        else:
+            continue
+        break
+
+# map suitable schedule
+if not test_schedule:
+    print(
+        f"WARNING: Could not find any schedule for '{test_file_name}'"
+    )
+    sys.exit(0)
+elif "isolation" in test_schedule:
+    test_schedule = 'base_isolation_schedule'
+elif "failure" in test_schedule:
+    test_schedule = 'failure_base_schedule'
+elif "mx" in test_schedule:
+    test_schedule = 'mx_base_schedule'
+else:
+    test_schedule = 'base_schedule'
```

Comment:
I think for flaky test detection base_schedule makes sense (the extra runtime won't matter much), but when testing locally you want minimal to be used instead by default. So let's make this a flag, like `--use-base-schedule`. The same is the case  for the mx schedule. During development you'll want `mx_minimal_schedule` by default.

PS. In future versions of this script I think the default should be made smarter. We should have a hardcoded set of tests that depend on the base_schedule instead of the minimal_schedule.

Reply:
I think we can use the `-s` parameter to override the schedule assignment in that case. Does it make sense?